csrc/quantization/cutlass_w8a8/grouped_mm_c3x.cuh

```diff
@@ -0,0 +1,188 @@
+#include "cutlass/epilogue/collective/collective_builder.hpp"
+#include "cutlass/gemm/device/gemm_universal_adapter.h"
+
+#include "cutlass_extensions/epilogue/scaled_mm_epilogues_c3x.hpp"
+#include "cutlass_extensions/common.hpp"
+
+using namespace cute;
+
+#if defined __CUDA_ARCH__ && __CUDA_ARCH__ >= 900
+  #define ENABLE_SM90_KERNEL_LEVEL 1
+#endif
+
+__global__ void get_group_gemm_starts(
+    int32_t* expert_offsets, int64_t* a_offsets, int64_t* b_offsets,
+    int64_t* out_offsets, int64_t* a_scales_offsets, int64_t* b_scales_offsets,
+    const int64_t a_base_as_int, const int64_t b_base_as_int,
+    const int64_t out_base_as_int, const int64_t a_scales_base_as_int,
+    const int64_t b_scales_base_as_int, int64_t n, int64_t k,
+    bool per_act_token, bool per_out_ch, int64_t ab_size, int64_t c_size,
+    int64_t acc_size) {
+  int expert_id = threadIdx.x;
+
+  int64_t expert_offset = expert_offsets[expert_id];
+
+  a_offsets[expert_id] = a_base_as_int + expert_offset * k * ab_size;
+  b_offsets[expert_id] = b_base_as_int + expert_id * k * n * ab_size;
+  out_offsets[expert_id] = out_base_as_int + expert_offset * n * c_size;
+  a_scales_offsets[expert_id] =
+      a_scales_base_as_int + (per_act_token ? expert_offset : 0) * acc_size;
+  b_scales_offsets[expert_id] =
+      b_scales_base_as_int +
+      (per_out_ch ? n * expert_id : expert_id) * acc_size;
+}
+
+namespace {
+
+using ProblemShape =
+    cutlass::gemm::GroupProblemShape<cute::Shape<int, int, int>>;
+using ElementAB_Type = cutlass::float_e4m3_t;
+using ElementC_Type = cutlass::half_t;
+
+using ElementAccumulator = float;
+using ArchTag = cutlass::arch::Sm90;
+using OperatorClass = cutlass::arch::OpClassTensorOp;
+
+using LayoutA = cutlass::layout::RowMajor;
+using LayoutB = cutlass::layout::ColumnMajor;
+using LayoutC = cutlass::layout::RowMajor;
+
+template <typename ElementAB_, typename ElementC_,
+          template <typename, typename, typename> typename Epilogue_,
+          typename TileShape, typename ClusterShape, typename KernelSchedule,
+          typename EpilogueSchedule>
+struct cutlass_3x_group_gemm {
+  using ElementAB = ElementAB_;
+  using ElementC = void;
+  using ElementD = ElementC_;
+  using ElementAccumulator = float;
+
+  using Epilogue = Epilogue_<ElementAccumulator, ElementD, TileShape>;
+
+  using StrideC =
+      cute::remove_pointer_t<cute::Stride<int64_t, cute::Int<1>, cute::Int<0>>>;
+
+  const int AlignmentAB = 128 / cutlass::sizeof_bits<ElementAB>::value;
+  const int AlignmentC = 128 / cutlass::sizeof_bits<ElementD>::value;
+
+  using EVTCompute = typename Epilogue::EVTCompute;
+
+  using CollectiveEpilogue =
+      typename cutlass::epilogue::collective::CollectiveBuilder<
+          ArchTag, OperatorClass, TileShape, ClusterShape,
+          cutlass::epilogue::collective::EpilogueTileAuto, ElementAccumulator,
+          ElementAccumulator, ElementC, LayoutC*, 4, ElementD, LayoutC*, 4,
+          EpilogueSchedule, EVTCompute>::CollectiveOp;
```

Comment:
Should these `4`s be `AlignmentC` instead? It looks like `AlignmentC` isn't used anywhere